genesis/src/lib.rs

```diff
@@ -28,16 +31,121 @@ pub static GENESIS_ACCOUNT_ID: Lazy<AccountId> =
     Lazy::new(|| AccountId::new(GENESIS_DOMAIN_ID.clone(), "genesis".parse().expect("Valid")));
 
 /// Genesis transaction
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Decode, Encode)]
 #[repr(transparent)]
 pub struct GenesisTransaction(pub SignedTransaction);
 
+impl GenesisTransaction {
+    /// Construct single genesis transaction with [`Executor`] and transactions unified under a single transaction
+    pub fn new_unified(
+        raw_block: RawGenesisBlock,
+        chain_id: &ChainId,
+        genesis_key_pair: &KeyPair,
+    ) -> GenesisTransaction {
+        Self(raw_block.unify().sign(chain_id.clone(), genesis_key_pair))
```

Comment:
Leaves allocation to the caller if they need to:

```suggestion
        chain_id: ChainId,
        genesis_key_pair: &KeyPair,
    ) -> GenesisTransaction {
        Self(raw_block.unify().sign(chain_id, genesis_key_pair))
```